### PR TITLE
Add google calendar action

### PR DESCRIPTION
**Pull request by Cleopha**
# Description

- Add google service in dispatcher
- Add calendar action: when an event starts in less than 10 minutes
- Remove useless fonction

@TomChv Please add New action type in gRPC and database 🙏🏼 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [ ] I have tested this code
- [X] I have added sufficient documentation in the code

### CODE REVIEW

File: obelix/idefix/services/google/client.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	Operator  *operator.IdefixOperator
+}
+
+// configure get credentials to connect to GitHub API
+func (c *Client) configure() error {
+	//TODO Credentials API
+	client, err := devauth.New(context.Background(), []string{
+		"https://www.googleapis.com/auth/bigquery",
+		"https://www.googleapis.com/auth/blogger",
+		"https://www.googleapis.com/auth/calendar",
```
Comment (0xpanoramix) [inline obelix/idefix/services/google/client.go]: Scopes should be contained in the `Client` structure.

File: obelix/idefix/services/google/client.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	srv, err := calendar.NewService(context.Background(), option.WithHTTPClient(c.Requester))
+	if err != nil {
+		log.Fatalf("Unable to retrieve Calendar client: %v", err)
+	}
+
+	err = gc.Parse(srv)
+	if err != nil || !errors.Is(err, ErrNoUpcomingEvent) {
+		return fmt.Errorf("failed to parse gcalendar data: %w", err)
+	}
```
Comment (0xpanoramix) [inline obelix/idefix/services/google/client.go]: Can you inline this ?

**Review by 0xpanoramix — CHANGES_REQUESTED** [review]
LGTM :rocket:

File: obelix/idefix/services/google/client.go (hunk 1 above)
Comment (Cleopha) [reply obelix/idefix/services/google/client.go]: 15e9a101b7deb41a36a6d1ed34f2529680d4230f

File: obelix/idefix/services/google/client.go (hunk 2 above)
Comment (Cleopha) [reply obelix/idefix/services/google/client.go]: 96ecf088b2cf205ee3e11412cc2a3c60073d6498

**Review by 0xpanoramix — APPROVED** [review]

File: obelix/idefix/services/google/client.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	Requester *http.Client
+	Scope     []string
+	Operator  *operator.IdefixOperator
+}
+
+// configure get credentials to connect to Google API
+func (c *Client) configure() error {
+	//TODO Credentials API
+	client, err := devauth.New(context.Background(), c.Scope)
+
```
Comment (TomChv) [inline obelix/idefix/services/google/client.go:26]: Blank line

File: obelix/idefix/services/google/client.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+func (c *Client) CalendarNearestEvent(taskID string, prm *structpb.Struct) error {
+	var gc GCalendar
+
+	if err := c.configure(); err != nil {
+		return fmt.Errorf("failed to configure github client: %w", err)
+	}
+
+	srv, err := calendar.NewService(context.Background(), option.WithHTTPClient(c.Requester))
+	if err != nil {
+		log.Fatalf("Unable to retrieve Calendar client: %v", err)
```
Comment (TomChv) [inline obelix/idefix/services/google/client.go]: Are you sure we want to exit program if we can't use one of our action?

**Review by TomChv — APPROVED** [review]
Few comments, LGTM

File: obelix/idefix/services/google/client.go (hunk 3 above)
Comment (0xpanoramix) [reply obelix/idefix/services/google/client.go:26]: The linter is configured that way ;)

File: obelix/idefix/services/google/client.go (hunk 4 above)
Comment (0xpanoramix) [reply obelix/idefix/services/google/client.go]: Nice catch ! Removed in https://github.com/Cleopha/ifttt-like/pull/55/commits/f4ec45d8b036b032e2eb1b7c4341eac9b18706f8

Comment (Cleopha) [reply obelix/idefix/services/google/client.go]: that's my boy

Comment (Cleopha) [reply obelix/idefix/services/google/client.go]: @PtitLuca please check the linter

**Review by TomChv — APPROVED** [review]
Good job, remove the label `dont-merge` and go

**Comment by Cleopha** [comment]
> Good job, remove the label `dont-merge` and go

no it's don't test ^^'

**Comment by 0xpanoramix** [comment]
@TomChv This PR is not mergeable for now because there is no such thing as `CALENDAR_NEAREST_EVENT` in the proto process (the protobuf definition and the backend services).
I'll wait for you to sync up with me to include this in the code.

**Review by 0xpanoramix — APPROVED** [review]

**Review by 0xpanoramix — CHANGES_REQUESTED** [review]
Please document your code before merging :rocket:

**Review by 0xpanoramix — APPROVED** [review]

**Pull request merged**